File: ui/src/services/api/gitService.ts
/**
 * Git operations API service
 */

import { apiGet, apiPost } from './api-client.js';
import type { GitStatus } from '../../types/domain.js';
import type { DiffData } from '../../types/api.js';

interface GitStatusResponse {
  status: GitStatus;
}

/**
 * Fetch Git status for a worktree
 */
export async function fetchGitStatus(
  org: string,
  repo: string,
  branch: string,
  entryLimit?: number,
  commitLimit?: number
): Promise<GitStatus | null> {
  const params = new URLSearchParams({ org, repo, branch });
  if (Number.isFinite(entryLimit)) {
    params.set('entryLimit', String(entryLimit));
  }
  if (Number.isFinite(commitLimit)) {
    params.set('commitLimit', String(commitLimit));
  }

  const response = await apiGet<GitStatusResponse>(
    `/api/git/status?${params.toString()}`,
    { errorPrefix: 'Failed to fetch git status' }
  );
  
  return response.status || null;
}

/**
 * Fetch diff for a file
 */
export async function fetchDiff(
  org: string,
  repo: string,
  branch: string,
  path: string,
  previousPath: string | null = null,
  mode = 'changes',
  status = ''
): Promise<DiffData> {
  const body = await apiPost<DiffData>(
    '/api/git/diff',
    {
      org,
      repo,
      branch,
      path,
      previousPath,
      mode,
      status,
    },
    { errorPrefix: 'Failed to load diff' }
  );

  const payload = body && typeof body === 'object' ? body : null;
  
  return {
    diff: payload && typeof payload.diff === 'string' ? payload.diff : '',
    path: payload?.path || path,
    previousPath: payload?.previousPath ?? previousPath,
    mode: payload?.mode || mode,
    status: payload?.status || status,
  };
}

File: mobile/ios/AgentrixMobile/Services/GitService.swift
import Foundation

final class GitService {
    private let api: AgentrixAPIClient

    init(api: AgentrixAPIClient) {
        self.api = api
    }

    func fetchStatus(org: String, repo: String, branch: String) async throws -> GitStatusResponse.Status {
        var components = URLComponents()
        components.queryItems = [
            URLQueryItem(name: "org", value: org),
            URLQueryItem(name: "repo", value: repo),
            URLQueryItem(name: "branch", value: branch),
            URLQueryItem(name: "entryLimit", value: "200"),
            URLQueryItem(name: "commitLimit", value: "20")
        ]
        let query = components.percentEncodedQuery.map { "?\($0)" } ?? ""
        let response: GitStatusResponse = try await api.request("/api/git/status\(query)")
        return response.status
    }

    func fetchDiff(
        org: String,
        repo: String,
        branch: String,
        path: String,
        previousPath: String?,
        mode: String?,
        status: String?
    ) async throws -> GitDiffResponse {
        struct Payload: Encodable {
            let org: String
            let repo: String
            let branch: String
            let path: String
            let mode: String?
            let previousPath: String?
            let status: String?
        }
        return try await api.request(
            "/api/git/diff",
            method: .post,
            body: Payload(
                org: org,
                repo: repo,
                branch: branch,
                path: path,
                mode: mode,
                previousPath: previousPath,
                status: status
            )
        )
    }
}
